src/pykeen/contrib/lightning.py

```diff
@@ -144,21 +144,112 @@ def configure_optimizers(self):
         )
 
 
-def _main():
+class SLCWALitModule(LitModule):
+    """A PyTorch Lightning module for training a model with sLCWA training loop."""
+
+    # docstr-coverage: inherited
+    def _step(self, batch, prefix: str):  # noqa: D102
+        positives, negatives, masks = batch
+        loss = self.loss.process_slcwa_scores(
+            positive_scores=self.model.score_hrt(hrt_batch=positives),
+            negative_scores=self.model.score_hrt(hrt_batch=negatives),
+            batch_filter=masks,
+            num_entities=self.model.num_entities,
+        )
+        self.log(f"{prefix}_loss", loss)
+        return loss
+
+    # docstr-coverage: inherited
+    def _dataloader(
+        self, triples_factory: CoreTriplesFactory, shuffle: bool = False
+    ) -> torch.utils.data.DataLoader:  # noqa: D102
+        return torch.utils.data.DataLoader(
+            dataset=triples_factory.create_slcwa_instances(),
+            batch_size=self.batch_size,
+            shuffle=shuffle,
+        )
+
+
+class LCWALitModule(LitModule):
+    """A PyTorch Lightning module for training a model with LCWA training loop.
+
+    .. seealso:: https://github.com/pykeen/pykeen/pull/905
+    """
+
+    # docstr-coverage: inherited
+    def _step(self, batch, prefix: str):  # noqa: D102
+        hr_batch, labels = batch
+        scores = self.model.score_t(hr_batch=hr_batch)
+        loss = self.loss.process_lcwa_scores(predictions=scores, labels=labels)
+        self.log(f"{prefix}_loss", loss)
+        return loss
+
+    # docstr-coverage: inherited
+    def _dataloader(
+        self, triples_factory: CoreTriplesFactory, shuffle: bool = False
+    ) -> torch.utils.data.DataLoader:  # noqa: D102
+        return torch.utils.data.DataLoader(
+            dataset=triples_factory.create_lcwa_instances(),
+            batch_size=self.batch_size,
+            shuffle=shuffle,
+        )
+
+
+# TODO: this does not seem to work
+# lit_module_resolver: ClassResolver[LitModule] = ClassResolver.from_subclasses(
+#     base=LitModule,
```

Comment:
maybe make an intermediate base class like `PyKEENLightningModule` to more easily hook into?

Reply:
`LitModule` is such a class

Reply:
oh my b! missed it in the diff